docs/runtime/go-metrics.md

```diff
@@ -0,0 +1,173 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: Go Runtime
+--->
+
+# Semantic Conventions for Go Runtime Metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+This document describes semantic conventions for Go runtime metrics in OpenTelemetry.
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [Go Memory](#go-memory)
+  - [Metric: `go.memory.used`](#metric-gomemoryused)
+  - [Metric: `go.memory.limit`](#metric-gomemorylimit)
+  - [Metric: `go.memory.allocated`](#metric-gomemoryallocated)
+  - [Metric: `go.memory.allocations`](#metric-gomemoryallocations)
+- [Go Garbage Collection](#go-garbage-collection)
+  - [Metric: `go.memory.gc.target`](#metric-gomemorygctarget)
+  - [Metric: `go.memory.gc.user_target`](#metric-gomemorygcuser_target)
+- [Go Threads](#go-threads)
+  - [Metric: `go.thread.limit`](#metric-gothreadlimit)
+- [Go Scheduler](#go-scheduler)
+  - [Metric: `go.schedule.duration`](#metric-goscheduleduration)
+
+<!-- tocstop -->
+
+## Go Memory
+
+**Description:** Go runtime metrics captured under the namespace `go.memory.*`
+
+### Metric: `go.memory.used`
+
+This metric is [recommended][MetricRecommended].
+This metric is obtained from Go's [`runtime/metrics`][RuntimeMetrics] package using `/memory/classes` metrics.
+
+<!-- semconv metric.go.memory.used(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.used` | UpDownCounter | `By` | Virtual memory mapped by the Go runtime. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.used(full) -->
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| `go.memory.type` | string | The type of memory. | `released`; `stack` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+`go.memory.type` MUST be one of the following:
+
+| Value  | Description | Stability |
+|---|---|---|
+| `released` | Memory that is completely free and has been returned to the underlying system. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `stack` | Memory allocated from the heap that is reserved for stack space, whether or not it is currently in-use. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `other` | All memory mapped by the Go runtime into the current process as read-write, excluding other categories of memory usage. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
I agree, the calculation of `other` seems under-specified. I'm assuming it is:

```
/memory/classes/total:bytes - /memory/classes/heap/released:bytes - /memory/classes/heap/stacks:bytes
```

If that's the case, I'm not sure `released` should be included as a memory.type here.

Reply:
I removed released memory from this metric, and split it out separately.  That means `go.memory.usage` is now defined as `/memory/classes/total:bytes - /memory/classes/heap/released:bytes`, which is all memory usage subject to `go.memory.limit` (based on https://go.dev/doc/gc-guide).  I think that makes using the two metrics intuitive.

As far as what is now included in `go.memory.usage`, [MemStats](https://pkg.go.dev/runtime#MemStats) seems to have the clearest description, since it is Sys - HeapReleased.  It includes the heap (excluding released), the stack (including OS thread stacks; excluding any stacks allocated by C code), mpsan structures, mcache structures, profiling bucket hash tables, gc metadata, and miscellaneous off-heap runtime allocations.